src/games/strategy/triplea/ai/proAI/data/ProTerritoryManager.java

```diff
@@ -1,31 +1,1144 @@
+  }
+
+  public void populateEnemyDefenseOptions() {
+    findScrambleOptions(player, attackOptions.getTerritoryMap());
+    enemyDefendOptions = findEnemyDefendOptions(player);
+  }
+
+  public List<ProTerritory> removeTerritoriesThatCantBeConquered() {
+    return removeTerritoriesThatCantBeConquered(player, attackOptions.getTerritoryMap(),
+        attackOptions.getUnitMoveMap(), attackOptions.getTransportMoveMap(), alliedAttackOptions, enemyDefendOptions,
+        false);
+  }
+
+  public List<ProTerritory> removePotentialTerritoriesThatCantBeConquered() {
+    return removeTerritoriesThatCantBeConquered(player, potentialAttackOptions.getTerritoryMap(),
+        potentialAttackOptions.getUnitMoveMap(), potentialAttackOptions.getTransportMoveMap(), alliedAttackOptions,
+        enemyDefendOptions, true);
+  }
+
+  public ProMyMoveOptions getAttackOptions() {
+    return attackOptions;
+  }
+
+  public ProMyMoveOptions getPotentialAttackOptions() {
+    return potentialAttackOptions;
+  }
+
+  public ProMyMoveOptions getDefendOptions() {
+    return defendOptions;
+  }
+
+  public ProOtherMoveOptions getAlliedAttackOptions() {
+    return alliedAttackOptions;
   }
 
-  public ProTerritory get(final Territory t) {
-    return territoryMap.get(t);
+  public ProOtherMoveOptions getEnemyDefendOptions() {
+    return enemyDefendOptions;
+  }
+
+  public ProOtherMoveOptions getEnemyAttackOptions() {
+    return enemyAttackOptions;
+  }
+
+  public List<Territory> getDefendTerritories() {
+    return new ArrayList<Territory>(defendOptions.getTerritoryMap().keySet());
   }
 
   public List<Territory> getStrafingTerritories() {
     final List<Territory> strafingTerritories = new ArrayList<Territory>();
-    for (final Territory t : territoryMap.keySet()) {
-      if (territoryMap.get(t).isStrafing()) {
+    for (final Territory t : attackOptions.getTerritoryMap().keySet()) {
+      if (attackOptions.getTerritoryMap().get(t).isStrafing()) {
         strafingTerritories.add(t);
       }
     }
     return strafingTerritories;
   }
 
+  public List<Territory> getCantHoldTerritories() {
+    final List<Territory> territoriesThatCantBeHeld = new ArrayList<Territory>();
+    for (final Territory t : defendOptions.getTerritoryMap().keySet()) {
+      if (!defendOptions.getTerritoryMap().get(t).isCanHold()) {
+        territoriesThatCantBeHeld.add(t);
+      }
+    }
+    return territoriesThatCantBeHeld;
+  }
+
+  public boolean haveUsedAllAttackTransports() {
+    final Set<Unit> movedTransports = new HashSet<Unit>();
+    for (final ProTerritory patd : attackOptions.getTerritoryMap().values()) {
+      movedTransports.addAll(patd.getAmphibAttackMap().keySet());
+      movedTransports.addAll(Match.getMatches(patd.getUnits(), Matches.UnitIsTransport));
+    }
+    return movedTransports.size() >= attackOptions.getTransportList().size();
+  }
+
+  private void findScrambleOptions(final PlayerID player, final Map<Territory, ProTerritory> moveMap) {
```

Comment:
Long method code smell, might be better breaking this method up so it contains the high level flow and calls a few  private helper methods for the lower level details.